### PR TITLE
add transform_error algorithm

### PR DESCRIPTION
**Pull request by ericniebler**

### CODE REVIEW

**Review by ispeters — APPROVED** [review]
lgtm

**Comment by ispeters** [comment]
You should edit the PR title.

File: include/unifex/transform_error.hpp [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+      }
+    }
+  }
+
+private:
+  template(typename CPO, typename Self)
+    (requires is_receiver_query_cpo_v<CPO> AND
+        same_as<remove_cvref_t<Self>, type> AND
+        is_callable_v<CPO, const Receiver&>)
+  friend auto tag_invoke(CPO cpo, Self&& r)
```
[the remark this answers is not in the document]
Comment (ericniebler) [reply include/unifex/transform_error.hpp:125]: `Self&&` is a forwarding reference. `type&&` would be an rvalue reference.

File: include/unifex/transform_error.hpp [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+    auto op = op_; // preserve pointer value.
+    if constexpr (
+      is_nothrow_callable_v<Error> &&
+      is_nothrow_connectable_v<final_sender_t, final_receiver>) {
+      op->startedOp_ = 0;
+      unifex::deactivate_union_member(op->sourceOp_);
+      unifex::activate_union_member_with(op->finalOp_, [&] {
+        return unifex::connect(std::move(op->error_)(), final_receiver{op});
+      });
+      op->startedOp_ = 0 - 1;
```
[the remark this answers is not in the document]
Comment (ericniebler) [reply include/unifex/transform_error.hpp:103]: Dunno. Copied from `transform_done`. @lewissbaker ?

**Pull request merged**